api/src/context.rs

```diff
@@ -453,11 +483,30 @@ impl Context {
         version: u64,
         limit: u64,
     ) -> Result<(Vec<(ModuleId, Vec<u8>)>, Option<StateKey>)> {
-        let account_iter = self.db.get_prefixed_state_value_iterator(
-            &StateKeyPrefix::from(address),
-            prev_state_key,
-            version,
-        )?;
+        let account_iter = if !db_sharding_enabled(&self.node_config) {
+            Box::new(
+                self.db
+                    .get_prefixed_state_value_iterator(
+                        &StateKeyPrefix::from(address),
+                        prev_state_key,
+                        version,
+                    )?
+                    .map(|item| item.map_err(|err| anyhow!(err.to_string()))),
+            )
+        } else {
+            ensure!(
+                self.node_config.indexer_db_config.enable_statekeys(),
+                "Internal statekeys indexer is not enabled"
+            );
```

Comment:
I think the indexer reader should carry the config itself and do the assertion internally.

Reply:
It is carried in IndexerReaders. But I want quick returns earlier if it fails the check. I can remove all these ensures